src/tweenable.rs

```diff
@@ -660,28 +660,26 @@ impl<T> Tweenable<T> for Sequence<T> {
 
     fn tick(
         &mut self,
-        delta: Duration,
+        mut delta: Duration,
         target: &mut T,
         entity: Entity,
         event_writer: &mut EventWriter<TweenCompleted>,
     ) -> TweenState {
-        if self.index < self.tweens.len() {
-            let mut state = TweenState::Active;
-            self.time = (self.time + delta).min(self.duration);
+        self.time = (self.time + delta).min(self.duration);
+        while self.index < self.tweens.len() {
             let tween = &mut self.tweens[self.index];
-            let tween_state = tween.tick(delta, target, entity, event_writer);
-            if tween_state == TweenState::Completed {
-                tween.rewind();
-                self.index += 1;
-                if self.index >= self.tweens.len() {
-                    state = TweenState::Completed;
-                    self.times_completed = 1;
-                }
+            let tween_remaining = tween.duration().mul_f32(1.0 - tween.progress());
```

Comment:
Can we avoid using `progress` and multiplications, and just use arithmetic on `Duration`s? We're trying to make progress a second-class concept. That's also probably slower, and has more chance to accumulate errors.

Reply:
Technically calculating `tween_remaining` is only needed for the first tween in the loop because the next tweens should all have `tween.progress() == 0`, so I could move it out of the loop.

Other options to calculate `tween_remaining`:

1. Subtract the `Duration`s of all the previous tweens from `self.time`.
2. Track the current tween's progress in a new field.

Option 1 avoids `progress()` but takes linear time. Option 2 avoids `progress()` and takes constant time, but makes `Sequence` a little more complex + uses some memory. Option 2 seems better to me.

Reply:
Ah sorry yes it's because we're missing #38, which I think is essentially option 2. I need to find some time to go review and merge those PRs so we have a better base to iterate on (and before we upgrade to 0.8).

Reply:
Let's leave the progress for now.